nmaipy/reference_code.py

```diff
@@ -78,7 +78,7 @@ def get_building_status(building_poly: Polygon, parcel_poly: Union[Polygon, Mult
     Get the status of a building polygon based on its relationship to the parcel polygon.
     """
     building_is_multiparcel, belongs_in, belongs_out, building_is_small = is_building_multiparcel(building_poly, parcel_poly)
-    is_keep = building_is_small or belongs_in
+    is_keep = building_is_small or belongs_in or not belongs_out
```

Comment:
Not sure if this is better?

```suggestion
    is_keep = building_is_small or belongs_in or building_poly.within(parcel_poly)
```

Reply:
So for clarity, current behaviour is:
* It's possible for a building to neither belong in, nor belong out - and be discarded (even if it has high fidelity and confidence).
* That corner case requires the building to be >30 sqm, and <4m in width at the widest point (3m max width after eroding 0.5m buffer around the edge) - so a 7.5m x 4m building would be just on the edge, and anything longer would be excluded.

<4m wide buildings won't be habitable - rough check it would have to be a little narrower than a freestanding single car garage, and at least a bit longer.

Comparing your two sugestions:

* First one says the building has to belong somewhere (and if it doesn't explicitly belong out, then we assign it in). You might have a building that is >50% outside the parcel, but doesn't quite meet criteria to belong out - I don't think we want to force it to belong in. We'd probably prefer to assign it to whichever side claims >50% of the area?
* Second one still leaves out a narrow building straddling an AOI boundary, even if it falls more within than without.

What if we go with:

`is_keep = building_is_small or belongs_in or (not belongs_in and not belongs_out and building_parts_within > building_parts_without)`

i.e. if it doesn't belong in or out explicitly, we keep it if the majority of it falls within the AOI?

Reply:
That can be simplified to just saying the ratio of object inside the parcel > 0.5, right? If so, yes, I agree.

Reply:
Simplified is:

`is_keep = building_is_small or belongs_in or (not belongs_out and building_parts_within > building_parts_without)`